src/test/java/com/fasterxml/jackson/core/write/BufferRecyclerPoolTest.java

```diff
@@ -0,0 +1,66 @@
+package com.fasterxml.jackson.core.write;
+
+import com.fasterxml.jackson.core.BaseTest;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.util.BufferRecyclerPool;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class BufferRecyclerPoolTest extends BaseTest
```

Comment:
It would be great to actually test recycling aspects: no need to use multi-threading, just create sequence of operations, starting with a single `JsonFactory` and single parser/generator, to allocate buffers, check that underlying `BufferRecycler` is properly acquired/released.

If an accessor needs to be added in parser/generator implementation for testing, that's fine.

This is a nice to have tho and just for basic sanity checking. No need to create complicated schemes.

Reply:
Done https://github.com/FasterXML/jackson-core/pull/1064/commits/53941560446ffddec4ac17b245884de7a557e474